src/utils/did.js

```diff
@@ -76,23 +196,64 @@ export function validateDockDIDSS58Identifier(identifier) {
 }
 
 /**
- * Gets the hexadecimal value of the given DID.
+ * Takes a DID string, gets the hexadecimal value of that and returns a `DockDidMethodKey` or `DockDid` object.
+ * @param {string} did -  The DID can be passed as fully qualified DID like `did:dock:<SS58 string>` or
+ * `did:key:<value>` or a 32 byte hex string
+ * @return {DockDidOrDidMethodKey} Returns a `DockDidMethodKey` or `DockDid` object.
+ */
+export function typedHexDID(did) {
+  const strDid = did.toString();
+  const hex = getHexIdentifier(
+    did,
+    [DockDIDQualifier, DockDIDMethodKeyQualifier],
+    DockDIDByteSize,
+  );
+
+  if (strDid.startsWith(DockDidMethodKeySecp256k1Prefix)) {
+    return new DockDidMethodKey(new PublicKeySecp256k1(hex));
+  } else if (strDid.startsWith(DockDidMethodKeyEd25519Prefix)) {
+    return new DockDidMethodKey(new PublicKeyEd25519(hex));
+  } else {
+    validateDockDIDHexIdentifier(hex);
+    return new DockDid(hex);
+  }
+}
+
+/**
+ * Gets the hexadecimal value of the given DID received from the substrate side.
  * @param {string} did -  The DID can be passed as fully qualified DID like `did:dock:<SS58 string>` or
  * a 32 byte hex string
  * @return {string} Returns the hexadecimal representation of the DID.
  */
-export function getHexIdentifierFromDID(did) {
-  return getHexIdentifier(did, DockDIDQualifier, validateDockDIDHexIdentifier, DockDIDByteSize);
+export function typedHexDIDFromSubstrate(did) {
+  const hex = getHexIdentifier(
+    u8aToHex(did.isDid ? did.asDid : did.asDidMethodKey),
+    [],
+    DockDIDByteSize,
+  );
+
+  if (did.isDid) {
+    return new DockDid(hex);
+  } else if (did.isDidMethodKey) {
+    return new DockDidMethodKey(hex);
```

Comment:
In `typedHexDID`, a `PublicKey` object is passed to `DockDidMethodKey` but here hex string is passed, why is that? What is the expectation from this function's returned value?